### ISSUE TITLE
Debugging overlay is hidden behind sensor housing / rounded corners on iPhone X

### ISSUE THREAD

**Issue by rhysforyou** [facebook/react-native#18141]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

```
Environment:
  OS: macOS High Sierra 10.13.3
  Node: 9.4.0
  Yarn: 1.3.2
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.1
  react-native: 0.54.0 => 0.54.0
```

### Steps to Reproduce

Bootstrap a new React Native app and run it in debug mode on an iPhone X

### Expected Behavior

The debugging overlay, including the bar that appears over the status bar to show loading progress, as well as the various warning messages that show up at the bottom of the screen, should not be obscured by the sensor housing or rounded edges of the iPhone X.

### Actual Behavior

The sensor housing on the iPhone X almost completely obscures the content of the 'loading' bar, and the home indicator and rounded corners on the bottom of the screen can slightly obscure the content of warning messages.

<img width="574" alt="screen shot 2018-03-01 at 11 10 04 am" src="https://user-images.githubusercontent.com/320910/36820285-287a054c-1d41-11e8-8d2a-0f4562bafdb8.png">

<img width="574" alt="screen shot 2018-03-01 at 11 11 11 am" src="https://user-images.githubusercontent.com/320910/36820328-675cbfe8-1d41-11e8-9b5a-08523d9d29bb.png">

### Reproducible Demo

This issue occurs on any React Native project, so I'm omitting a specific demo.

**Comment by rhysforyou** [comment]
Yeah, I tried updating to the latest React Native release but no dice.

**Comment by hramos** [comment]
Hey! Can you edit your issue and make sure it specifies the exact version you tried this on? At this point, "latest" could be 0.53 or 0.54 depending on how closely you watch our releases, so we really want to be explicit here :)

**Comment by rhysforyou** [comment]
Done, I've updated the issue

**Comment by aledalgrande** [comment]
Happening for me too on this setup:
```
Environment:
  OS: macOS High Sierra 10.13.4
  Node: 9.9.0
  Yarn: 1.5.1
  npm: 5.6.0
  Watchman: 4.7.0
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: Not Found

Packages: (wanted => installed)
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.3
  react-native: 0.54.2 => 0.54.2
```

**Comment by TheSavior** [comment]
The yellowbox spacing issue got fixed in 0.56 rc2. I haven't experienced the loading bar being obscured and I do a bunch of work on an iPhone X. Can you see if that one reproes in a new project with the latest RC?

**Comment by rhysforyou** [comment]
Looks to be fixed 🎉
